include/iterlib/DifferenceIterator.h

```diff
@@ -0,0 +1,33 @@
+#pragma once
+
+#include "iterlib/Iterator.h"
+
+namespace iterlib {
+
+class DifferenceIterator : public CompositeIterator {
+public:
+  DifferenceIterator(IteratorVector& children)
```

Comment:
Probably worthwhile to identify here that the children is required to be of size 2